magma_smaht/create_metawfr.py

```diff
@@ -140,35 +146,134 @@ def mwfr_ont_alignment(fileset_accession, smaht_key):
+        if not current_ref_genome:
+            raise Exception(f"File {file[UUID]} has not reference genome")
+        if reference_genome_uuid and current_ref_genome[UUID] != reference_genome_uuid:
+            raise Exception(f"Multiple reference genomes detected.")
+        reference_genome_uuid = current_ref_genome[UUID]
+    reference_genome_item = ff_utils.get_metadata(reference_genome_uuid, smaht_key)
+    reference_genome_file = reference_genome_item["files"][0][UUID]
+    reference_genome = [{"file": reference_genome_file}]
+    mwfr_input = [
+        get_mwfr_file_input_arg(INPUT_FILES_CRAM, crams),
+        get_mwfr_file_input_arg(GENOME_REFERENCE_FASTA, reference_genome),
+    ]
+    # pprint.pprint(mwfr_input)
+    create_and_post_mwfr(mwf[UUID], file_set, INPUT_FILES_CRAM, mwfr_input, smaht_key)
+
+
+################################################
+#   QC MetaWorkflowRuns
+################################################
+
+
 def mwfr_fastqc(fileset_accession, smaht_key):
 
     file_set = get_file_set(fileset_accession, smaht_key)
     mwf = get_latest_mwf(MWF_NAME_FASTQC, smaht_key)
     print(f"Using MetaWorkflow {mwf[ACCESSION]} ({mwf[ALIASES][0]})")
 
     # Get unaligned R2 reads in the fileset that don't have already QC
-    search_filter = f"?file_sets.uuid={file_set[UUID]}&type=UnalignedReads&file_format.display_title=fastq_gz&read_pair_number=R2&quality_metrics=No+value"
-    
-    files_to_run_r2 = ff_utils.search_metadata((f"search/{search_filter}"), key=smaht_key)
+    search_filter = f"?file_sets.uuid={file_set[UUID]}&type=File&file_format.display_title=fastq_gz&read_pair_number=R2&quality_metrics=No+value"
+
+    files_to_run_r2 = ff_utils.search_metadata(
+        (f"search/{search_filter}"), key=smaht_key
+    )
     files_to_run_r2.reverse()
 
     if len(files_to_run_r2) == 0:
         print(f"No files to run for search {search_filter}")
         return
-    
+
     # Create files list for input args
     files_input = []
     for dim, file_r2 in enumerate(files_to_run_r2):
         files_input.append({"file": file_r2[UUID], "dimension": f"1,{dim}"})
-        files_input.append({"file": file_r2["paired_with"][UUID], "dimension": f"0,{dim}"})
-    files_input_list = sorted(files_input, key=lambda x: x['dimension'])
+        files_input.append(
+            {"file": file_r2["paired_with"][UUID], "dimension": f"0,{dim}"}
+        )
+
+    # # Manual override
+    # files_input = []
+    # files_input.append({"file": "577b4259-81f5-4b0c-9ffc-254696b37493", "dimension": f"1,0"}) # R2
+    # files_input.append({"file": "cc0cb991-5fbc-4166-ad03-eaf8a42de649", "dimension": f"0,0"}) # R1
+
+    files_input_list = sorted(files_input, key=lambda x: x["dimension"])
 
     mwfr_input = [get_mwfr_file_input_arg(INPUT_FILES_FASTQ_GZ, files_input_list)]
     create_and_post_mwfr(
         mwf[UUID], file_set, INPUT_FILES_FASTQ_GZ, mwfr_input, smaht_key
     )
 
 
+def mwfr_fastq_qc_long_read(fileset_accession, smaht_key):
```

Comment:
Minor note: It's a little confusing having this function have fastq in the name when its getting bams

Reply:
Good point. Fixed.